pkgs/by-name/co/compiler-explorer/package.nix

```diff
@@ -0,0 +1,144 @@
+{ lib
+, stdenv
+, runCommand
+, writeText
+, buildNpmPackage
+, fetchFromGitHub
+, makeWrapper
+, python3Full
+, rsync
+, nodejs_20
+, husky
+, typescript
+, cmake
+, coreutils
+, gcc48, gcc49, gcc6, gcc7, gcc8, gcc9, gcc10, gcc11, gcc12, gcc13, clang_9
+, clang_11, clang_12, clang_13, clang_14, clang_15, clang_16, clang_17
+, lld_17, llvmPackages_17
+, rustc, rustfilt
+, gccgo13
+}:
+let
+
+  utils = import ./cfg/utils.nix { inherit lib writeText runCommand python3Full; };
+  defaultGcc = gcc13;
+  defaultClang = clang_17;
+
+  cppConfigFile = import ./cfg/c-cpp.nix {
+    inherit writeText utils defaultGcc defaultClang gcc48 gcc49 gcc6 gcc7 gcc8 gcc9 gcc10 gcc11 gcc12 gcc13 clang_9
+      clang_11 clang_12 clang_13 clang_14 clang_15 clang_16 clang_17 lld_17 llvmPackages_17;
+    isC = false;
+  };
+
+  cConfigFile = import ./cfg/c-cpp.nix {
+    inherit writeText utils defaultGcc defaultClang gcc48 gcc49 gcc6 gcc7 gcc8 gcc9 gcc10 gcc11 gcc12 gcc13 clang_9
+      clang_11 clang_12 clang_13 clang_14 clang_15 clang_16 clang_17 lld_17 llvmPackages_17;
+    isC = true;
+  };
+
+  rustConfigFile = import ./cfg/rust.nix {
+    inherit defaultGcc rustc rustfilt writeText utils;
+  };
+
+  goConfigFile = import ./cfg/go.nix {
+    inherit defaultGcc gccgo13 writeText utils;
+  };
+
+  pythonConfigFile = import ./cfg/python3.nix {
+    inherit defaultGcc python3Full writeText utils;
+  };
+
+  ceConfigFile = import ./cfg/compiler-explorer.nix {
+    inherit lib writeText utils python3Full defaultGcc cmake coreutils;
+  };
+  executionConfigFile = import ./cfg/compiler-explorer-execution.nix {
+    inherit writeText utils;
+  };
+
+in
+buildNpmPackage rec {
+
+  pname = "compiler-explorer";
+  version = "10723";
+  src = fetchFromGitHub {
+    owner = "compiler-explorer";
+    repo = "compiler-explorer";
+    rev = "gh-${version}";
+    hash = "sha256-e8G1WNII61FWGscxSTlB2GvDP5e3945AslDP2h4jr2k=";
+  };
+  npmDepsHash = "sha256-8e8JtVeJXx1NxwYlN4SRJB2K/RJv9plnAwlRNWHWD1M=";
+
+  nativeBuildInputs = [ nodejs_20 typescript husky rsync ];
```

Comment:
what's husky used for?

Reply:
It was required while building godbolt, if it was not there it gave an error, I can try removing and seeing what happens, since I did a lot of trial-error.